May 2020 Updates

francescobrivio opened the pull request:
***DNN update***
Updated DNN version to be used with training "2020-05-18-2".
  - Need to do a git pull in "cms_runII_dnn_models"
  - Need to do a checkout in "cms_hh_proc_interface" (README updated)
  - `DNNKLUBinterface`, `skimNtuple201*.cpp` and `skimOutputter.cpp` have been updated

***New Branches***
The new DNN requires some new branches which I added in `smalTree` and `bigTree`.
  - _CvsL_ and _CvsB_ taggers (defined [here for 2018](https://twiki.cern.ch/twiki/bin/viewauth/CMS/BtagRecommendation102X#Supported_Algorithms_and_Operati), 2016 and 2017 are the same) for b-jets and VBF-jets
  - _HHbtag_ values for b-jets and VBF-jets: these values for now are dummy, we will update them once we implement the HHbtag package

***skimUtils.h***
Added one header file `interface/skimUtils.h` which now contains all the functions that were repeated multiple times in the skimmers. The skimmers have consequently been cleaned of these functions. Having all the functions in one place should make them more easily mantainable.
  - Functions `turnOnCB` and `turnOnSF` were a residual from 2016 analysis and not used anymore: I deleted them

francescobrivio commented:
@camendola @dzuolo this is a quite large PR, I'm testing it and so far everything seems to be working fine, but a double check is needed.

dzuolo commented:
Ciao @francescobrivio I cross checked the association between variable and names in all the files and everything is ok to me!

francescobrivio commented:
@dzuolo could you have a look at this last commit e5e7df8 ? These are the fix that we discussed offline

francescobrivio commented:
Trigger updates:
- removed the ETau cross trigger SF in 2016 since no ETau cross trigger is used in 2016
- updated triggers prescale SF values according to https://twiki.cern.ch/twiki/bin/viewauth/CMS/DoubleHiggsToBBTauTauWorkingLegacyRun2#Triggers
@camendola this is what we discussed offline, please have a look

camendola commented on test/skimNtuple2018.cpp:
```diff
@@ -3791,7 +3791,7 @@ int main (int argc, char** argv)
                   else                                  SFTau2 = 0.97;
 
-                  float VBFtrigNorm = 1.;   // No need to normalize for 2018
+                  float VBFtrigNorm = 0.990342;   // from: https://twiki.cern.ch/twiki/bin/viewauth/CMS/DoubleHiggsToBBTauTauWorkingLegacyRun2#Triggers
```
mmm boh

francescobrivio replied:
yeah i'm not 100% sure either, I took it from the twiki. If you prefer I can put it back to 1. It shouldn't change much anyway

francescobrivio commented:
Just for book keeping: updated debug workflow for 2016 and 2017 as in #127

francescobrivio commented:
@camendola @dzuolo since you already reviewed I will merge this PR.
Remember that in order to use the "2020-05-18-2" DNN training you have to update the skim config files that I didn't touch.
